### DISCUSSION TITLE
#Beginner's Guide

### DISCUSSION THREAD

**Discussion by DuajDiaz** [DuajDiaz/Travel-Website#19] [Q&A] [answered]
What’s the Best Programming Language to Learn First?

**Comment by Bunny77K9** [comment, marked as answer]
It Depends...

Whether you’re looking to begin coding as a hobby, a new career, or just to enhance your current role, the first thing you’ll have to do is decide which programming language you want to start with.

There is no right answer, of course. Choosing a first language will depend on what kind of projects you want to work on, who you want to work for, or how easy you want it to be. Hopefully, this guide will help give you a better idea of which one you should pursue.

<h3>Python</h3>
Python is always recommended if you’re looking for an easy and even fun programming language to learn first. Rather than having to jump into strict syntax rules, Python reads like English and is simple to understand for someone who’s new to programming. This allows you to obtain a basic knowledge of coding practices without having to obsess over smaller details that are often important in other languages.

Python also is ideal for web development, graphic user interfaces (GUIs), and software development. In fact, it was used to build Instagram, YouTube, and Spotify, so it’s clearly in demand among employers in addition to having a faster onboarding.

Though it has its advantages, Python is often thought of as a slow language that requires more testing and is not as practical for developing mobile apps as other languages.

<h3>C#</h3>
While C is one of the more difficult languages to learn, it’s still an excellent first language pick up because almost all programming languages are implemented in it. This means that once you learn C, it’ll be simple to learn more languages like C++ and C#.

Because C is more “machine-level”, learning is great for teaching you how a computer functions. Software Developer Joel Spolsky compares it to understanding basic anatomy before becoming a medical doctor, making it the best way to code efficiently.

In this way, C is an exceptional choice to become a master coder and a talented developer from the get-go if you’re willing to take on the challenge.

<h3>Java</h3>
Java is an object-oriented and feature-heavy programming language that’s in high demand. It’s been built under the premise of “Write once, run anywhere,” meaning that it can be written on any device and work cross-platform.

This makes it one of the most desired (yes, we mean high-paid) language skills. So, if you’re looking to learn a language that’s going to get you a great career, this might be the one, especially since top employers for Java programmers include eBay, Amazon, and IBM.

Additionally, Java is often used for Android and iOS app development, as it’s the basis of the Android operating system, which makes it one of the best choices if you want to build mobile apps.

While it may not be as easy to pick up as Python, Java is a high-level language, and so it’s still relatively beginner-friendly. However, it has a slow startup and will take beginners much longer to deploy their first project.

**Reply by DuajDiaz** [reply]
Thank you Nandun!
